internals/overlord/logstate/loki/loki.go

```diff
@@ -84,6 +85,10 @@ func fillDefaultOptions(options *ClientOptions) *ClientOptions {
 	return options
 }
 
+func (c *Client) SetLabels(serviceName string, labels map[string]string) {
+	c.labels[serviceName] = labels
```

Comment:
I saw you previously did a `delete` if `labels` was nil. Isn't that better (in terms of memory reuse), as otherwise we'll keep the old map entries around?